src/cloudWatchLogs/changeLogSearch.ts

```diff
@@ -73,19 +75,21 @@ export async function changeLogSearchParams(
     param: 'filterPattern' | 'timeFilter'
 ): Promise<void> {
     let result: telemetry.Result = 'Succeeded'
-
     const oldUri = getActiveDocumentUri(registry)
-    const newData = await getNewData(registry, param, oldUri)
+
+    const oldData = registry.getLogData(oldUri)
+    if (!oldData) {
+        throw new Error(`cwl: Unable to find data for active URI ${oldUri}`)
+    }
+    const newData = await getNewData(param, oldData)
 
     if (!newData) {
         result = 'Cancelled'
         return
     }
 
-    registry.deregisterLog(oldUri)
     const newUri = createURIFromArgs(newData.logGroupInfo, newData.parameters)
 
     result = await prepareDocument(newUri, newData, registry)
-
     console.log(result)
```

Comment:
Make sure to delete these random `console.log` calls. It's ok for development but otherwise it's noise.

Reply:
Yeah, I have it in as a placeholder for telemetry, but I can delete them now?

Reply:
Oh I see. Yeah just delete them and leave comments instead:

```ts
// TODO: add telemetry ...
```